core/src/main/java/org/elasticsearch/index/search/MatchQuery.java

```diff
@@ -286,6 +286,8 @@ protected final Query termQuery(MappedFieldType fieldType, Object value, boolean
     }
 
     protected Query zeroTermsQuery() {
+        // TODO This is weird: DEFAULT_ZERO_TERMS_QUERY is a public static final field, so this can have only one value.
+        // why are we then having this if clause in here?
```

Comment:
After re-checking what's going on I realised I missed the setter for zeroTermsQuery when reading this first. Removed the question.